src/hints/block_context.rs

```diff
@@ -286,3 +392,47 @@ pub fn get_block_mapping(
     };
     insert_value_from_var_name("state_entry", val, vm, ids_data, ap_tracking)
 }
+
+pub const ELEMENTS_GE_10: &str = "memory[ap] = to_felt_or_relocatable(ids.elements_end - ids.elements >= 10)";
+pub fn elements_ge_10(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let elements_end = get_ptr_from_var_name("elements_end", vm, ids_data, _ap_tracking)?;
+    let elements = get_ptr_from_var_name("elements", vm, ids_data, _ap_tracking)?;
+    insert_value_into_ap(vm, if (elements_end - elements)? >= 10 { Felt252::ONE } else { Felt252::ZERO })
+}
+
+pub const ELEMENTS_GE_2: &str = "memory[ap] = to_felt_or_relocatable(ids.elements_end - ids.elements >= 2)";
+pub fn elements_ge_2(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    _ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let elements_end = get_ptr_from_var_name("elements_end", vm, ids_data, _ap_tracking)?;
+    let elements = get_ptr_from_var_name("elements", vm, ids_data, _ap_tracking)?;
+    insert_value_into_ap(vm, if (elements_end - elements)? >= 2 { Felt252::ONE } else { Felt252::ZERO })
+}
+
+pub const IS_LEAF: &str = indoc! {r#"
+    from starkware.starknet.core.os.contract_class.compiled_class_hash_objects import (
+        BytecodeLeaf,
+    )
+    ids.is_leaf = 1 if isinstance(bytecode_segment_structure, BytecodeLeaf) else 0"#
+};
+pub fn is_leaf(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    // let bytecode_segment_structure = get_ptr_from_var_name("bytecode_segment_structure", vm,
+    // ids_data, ap_tracking)?; TODO: complete when bytecode_segment_structure is implemented
+    insert_value_from_var_name("is_leaf", 1, vm, ids_data, ap_tracking)
+}
```

Comment:
I think there should be unit tests for each hint that is already fully implemented. Implementing tests for all the hints at the end is gonna be such a pain otherwise.

Reply:
I would say that reasonable middle ground is this: If hints are straightforward there is no need for unit tests since this code will be executed by integration tests. I would write unit tests only for non trivial hints involving ifs/loops/complex data structures.